csrc/scheduler/matmul_utils.cpp

```diff
@@ -126,11 +126,19 @@ inline bool initCoreHeuristics(
   {
     // NOTE: compilation errors when async is enabled on Turing devices
     if (isAmpere(params->mma_macro)) {
-      constexpr int stages = 3;
-
-      params->circular_buffer_options.circular_buffer_smem_write = true;
+      // Find the number of circular buffer stages so that the entire pipeline
+      // is filled.
+      //
+      // The axes of the mma tensorviews are permuted to [B, M, N, K],
+      // so K / cta_tile_k is the circular buffer axis for both operands.
+      int64_t k_stages =
+          ceilDiv(problem_shape[(size_t)MatmulDimRole::K], cta_tile.k);
```

Comment:
@jacobhinkle Is this correct if we have split-k?

Reply:
No, I believe the numerator should be `ceilDiv(problem_shape[(size_t)MatmulDimRole::K], params.splitk_factor)` here. Good catch.

Reply:
I added this ^^^ to `limitCircularBufferingSmemOperands`, which updates the parameters from `matmul_heuristic_plugin` and `initCoreHeuristics`.

Can `splitk_factor > 1` in the `initCoreHeuristics`?

Reply:
It should not be `>1` in `initCoreHeuristics` at this time.